posts/Yoann/2019-04-05-hibernate-search-6-0-0-Alpha4.adoc

```diff
@@ -0,0 +1,149 @@
+    .predicate(f -> f.exists().onField("someField"))
+    .toQuery();
+----
+
+=== The Elasticsearch AWS integration is back
+
+We restored the AWS integration as part of https://hibernate.atlassian.net/browse/HSEARCH-3051[HSEARCH-3051].
+See https://docs.jboss.org/hibernate/search/6.0/reference/en-US/html_single/#backend-elasticsearch-configuration-aws[the dedicated section in the documentation]
+to configure your backend to talk with an AWS-managed Elasticsearch service.
+
+=== Converters can be bypassed in predicates
+
+DSL and projection converters automatically convert values from the type expected by the user to the underlying type exposed by backends.
+For example `java.util.Date` properties have a converter between that type and `Instant`.
+
+Just like the `.ignoreFieldBridge` option in Search 5,
+https://hibernate.atlassian.net/browse/HSEARCH-3510[there is now an option] to bypass these converters in Search 6:
+
+[source, JAVA, indent=0]
+----
+// When the user input is a java.util.Date
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.match().onField("myJavaUtilDate").matching(javaUtilDateUserInput))
+    .toQuery();
+// When the user input is a java.time.Instant
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.match().onField("myJavaUtilDate").matching(instantUserInput, DslConverter.DISABLED))
+    .toQuery();
+----
+
+[source, JAVA, indent=0]
+----
+// When we need the projection as a java.util.Date
+SearchQuery<Date> query = searchSession.search(MyEntity.class)
+    .asProjection(f -> f.field("myJavaUtilDate", Date.class)
+    .predicate(f -> f.matchAll())
+    .toQuery();
+
+
+// When we need the projection as a java.time.Instant
+SearchQuery<Instant> query = searchSession.search(MyEntity.class)
+    .asProjection(f -> f.field("myJavaUtilDate", Instant.class, ProjectionConverter.DISABLED)
+    .predicate(f -> f.matchAll())
+    .toQuery();
+----
+
+=== Analyzers can be overridden at query time
+
+Analyzers could already be overridden in Search 5, but only at the `QueryBuilder` level and on a per-field basis,
+which could be annoying if you had many fields in your predicates or if you needed two predicates with different overrides.
+
+In Search 6, https://hibernate.atlassian.net/browse/HSEARCH-3312[you can now override analyzers on a per-predicate basis]:
+
+[source, JAVA, indent=0]
+----
+SearchQuery<MyEntity> query = searchSession.search(MyEntity.class).asEntity()
+    .predicate(f -> f.match().onField("name_autocomplete").matching(javaUtilDateUserInput).analyzer("edgengram_query"))
+    .toQuery();
+----
+
+=== Configuration
+
+Since https://hibernate.atlassian.net/browse/HSEARCH-3521[HSEARCH-3521],
+the property keys for index-scoped configuration properties are now prefixed with `.backends.<backend name>`:
+for example `hibernate.search.backends.myBackend.indexes.myIndex.lifecycle.strategy = create`.
```

Comment:
`for example` appears stick to the next word.
I would put ``hibernate.search.backends.myBackend.indexes.myIndex.lifecycle.strategy = create`.` inside a quotation or code block.